Sources/StreamVideoSwiftUI/Utils/VideoRendererPool/VideoRendererPool.swift

```diff
@@ -0,0 +1,59 @@
+//
+// Copyright © 2024 Stream.io Inc. All rights reserved.
+//
+
+import Combine
+import Foundation
+import StreamVideo
+
+/// A pool for managing reusable `VideoRenderer` instances.
+final class VideoRendererPool {
+    /// The underlying pool of `VideoRenderer` instances managed by this pool.
+    private let pool: ReusePool<VideoRenderer>
+    /// A cancellable object to observe call end notifications for releasing all renderers.
+    private var callEndedCancellable: AnyCancellable?
+
+    /// Initializes the `VideoRendererPool` with a specified initial capacity.
+    ///
+    /// - Parameter initialCapacity: The initial capacity of the pool (default is 10).
+    init(initialCapacity: Int = 10) {
+        // Initialize the pool with a capacity and a factory closure to create `VideoRenderer` instances
+        pool = ReusePool(initialCapacity: initialCapacity) {
+            VideoRenderer(frame: CGRect(origin: .zero, size: .zero))
+        }
+
+        // Observe call end notifications to release all renderers when a call ends
+        callEndedCancellable = NotificationCenter.default.publisher(for: Notification.Name(CallNotification.callEnded))
+            .sink { [weak self] _ in
+                self?.pool.releaseAll()
```

Comment:
If we had a call with 100 people, this will grow to 100 views, and they won't be removed for the next calls? (e.g. with only few people)

Reply:
Actually the will be removed. `releaseAll` will maintain a number VideoRenders up to `initialCapacity` value and everything else will be released. In the scenario you mentioned, 10 VideoRenders will be marked as available and 90 will be released

Reply:
Cool, missed that part 👍